core/src/net/tcp_stream.cpp

```diff
@@ -0,0 +1,167 @@
+#ifdef _WIN32
+#else
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#endif
+
+#include <cstring>
+
+#include <cubos/core/log.hpp>
+#include <cubos/core/net/tcp_stream.hpp>
+#include <cubos/core/reflection/external/cstring.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+
+using cubos::core::net::Address;
+using cubos::core::net::TcpStream;
+
+bool TcpStream::connect(const Address& address, uint16_t port)
+{
+    mSock = socket(AF_INET, SOCK_STREAM, 0);
+    if (mSock == INVALID_SOCKET)
+    {
+        CUBOS_ERROR("Failed to allocate TCP socket: {}", logSystemError());
+        return false;
+    }
+
+    sockaddr_in addr;
+    addr.sin_family = AF_INET;
+    addr.sin_addr.s_addr = inet_addr(address.toString().c_str());
+    addr.sin_port = htons(port);
+
+    if (::connect(mSock, (struct sockaddr*)&addr, sizeof(addr)) < 0)
+    {
+        CUBOS_ERROR("Failed to connect TCP stream to {}:{}: {}", address.toString(), port, logSystemError());
+        return false;
+    }
+
+    CUBOS_INFO("New TCP stream at address {} and port '{}'", address.toString(), port);
+    return true;
+}
+
+bool TcpStream::inner(SOCKET socket)
+{
+    disconnect();
+
+    mSock = socket;
+    CUBOS_INFO("New TCP stream at socket {}", mSock);
+    return mSock != INVALID_SOCKET;
+}
```

Comment:
Why check if it is invalid here? I think this function should return `void`. The user is responsible for whatever they throw inside it, its not necessary to do the check for them.